### ISSUE TITLE
unused variables in list comprehension not reported

### ISSUE THREAD

**Issue by lilydjwg** [PyCQA/pyflakes#593]
Test code:

```py
def t(items):
  return ['{x}' for x in items]
```

There is a missing `f` prefix which makes the local `x` variable unused, but pyflakes doesn't notice this.

```
$ pyflakes --version
2.2.0 Python 3.8.6 on Linux
```

**Comment by asottile** [comment]
I agree this particular example should probably produce an error, but pyflakes is a bit timid about assignments in unpackings

for example, should this produce a warning (pyflakes currently assumes any sort of unpacking can introduce variables that don't need to be referenced)?

```python
return [f'{x} for x, y in items]
```

**Comment by lilydjwg** [comment]
Aren't unused names expected to be prefixed with `_`? If not, maybe pyflakes can detect missing `f`-prefixes in another way?

This kind of errors is not easy to notice otherwise: the types check with any issue; the `{x}` syntax inside normal strings is also special (for `.format`) so it's highlighted anyway.

**Comment by asottile** [comment]
that's not something that's errorable either

consider this valid snippet:

```python
FMT_S = 'foo {bar}'

def make_thing():
    return FMT_S.format(bar=...)
```

(of course this is a silly example that could be inlined, but it is still a case that comes up).  this would introduce a high-false-positive rule which isn't really appropriate for pyflakes and is better left to a flake8 plugin or pylint
